packages/firebase/index.d.ts

```diff
@@ -7768,10 +7768,23 @@ declare namespace firebase.firestore {
      * shared execution of queries and latency-compensated local document updates
      * across all connected instances.
      *
-     * @deprecated This setting is deprecated. To enabled synchronization between
+     * @deprecated This setting is deprecated. To enable synchronization between
      * multiple tabs, please use `synchronizeTabs: true` instead.
      */
     experimentalTabSynchronization?: boolean;
+
+    /**
+     * Whether to force enable persistence for the client. This cannot be used
+     * with `synchronizeTabs:true` and is primarily intended for use with Web
+     * Workers. Setting this to 'true' will enable persistence, but cause other
+     * tabs using persistence to fail.
+     *
+     * This setting may be removed in a future release. If you find yourself
+     * using it for a specific use case or run into any issues, please tell us
+     * about it in
+     * https://github.com/firebase/firebase-js-sdk/issues/983.
+     */
+    experimentalForceOwningTab?: boolean;
```

Comment:
The use case might be to pin an owner tab. Not sure why that might be useful, but it would avoid primary to secondary tab transition which AFAIK haven't been a problem.

As for the name:

To me, `WebWorker` is the umbrella term, but I might be mistaken (and propose `experimentalBackgroundWorkerPersistence` instead).

My main problem with the API name is that `experimentalForceOwningTab` only says what it does but doesn't say why it does it or what feature it unblocks. The fact that it enables WebWorker/SharedWorker/ServiceWorker persistence is not obvious from the name, and it isn't even document in the API description. The only place where it shows up is in the title of the issue that is linked. A user that wants to use Firestore in a WebWorker has no obvious way to discover that this is the setting they need to enable.

If you are happy with the name, feel free to ignore. This has been discussed for quite some time.